kernel/kernel-default/src/main/java/io/delta/kernel/data/vector/DefaultDecimalVector.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.data.vector;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.types.DataType;
+import io.delta.kernel.types.DecimalType;
+import static io.delta.kernel.DefaultKernelUtils.checkArgument;
+
+/**
+ * {@link io.delta.kernel.data.ColumnVector} implementation for decimal type data.
+ */
+public class DefaultDecimalVector extends AbstractColumnVector {
+
+    private final BigDecimal[] values;
+
+    /**
+     * Create an instance of {@link io.delta.kernel.data.ColumnVector} for decimal type.
+     *
+     * @param size number of elements in the vector.
+     * @param nullability Optional array of nullability value for each element in the vector.
```

Comment:
is the nullability vector needed, given the values array is an object array?

Reply:
removed it